File: problems/problems_624/solution.go
package problem624

import (
	"encoding/json"
	"log"
	"math"
	"strings"
)

func maxDistance(arrays [][]int) (ans int) {
	maxVal, minVal := arrays[0][len(arrays[0])-1], arrays[0][0]
	for i := 1; i < len(arrays); i++ {
		minCur, maxCur := arrays[i][0], arrays[i][len(arrays[i])-1]
		ans = max(ans, int(math.Abs(float64(maxVal-minCur))), int(math.Abs(float64(maxCur-minVal))))
		maxVal = max(maxVal, maxCur)
		minVal = min(minVal, minCur)
	}
	return
}

func Solve(inputJsonValues string) interface{} {
	inputValues := strings.Split(inputJsonValues, "\n")
	var arrays [][]int

	if err := json.Unmarshal([]byte(inputValues[0]), &arrays); err != nil {
		log.Fatal(err)
	}

	return maxDistance(arrays)
}

File: problems/problems_624/solution.py
import solution
from typing import *
from math import inf

class Solution(solution.Solution):
    def solve(self, test_input=None):
        return self.maxDistance(test_input)

    def maxDistance(self, arrays: List[List[int]]) -> int:
        min_val, max_val = inf, -inf
        ans = 0
        for arr in arrays:
            ans = max(ans, max(arr) - min_val, max_val - min(arr))
            min_val = min(min_val, arr[0])
            max_val = max(max_val, arr[-1])
        return ans
